pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -182,16 +182,23 @@ func (b *BridgePodInterface) discoverPodNetworkInterface() error {
 }
 
 func (b *BridgePodInterface) preparePodNetworkInterfaces() error {
+	delegateIp := true
+	if b.iface.Bridge.DelegateIp != nil {
+		delegateIp = *b.iface.Bridge.DelegateIp
+	}
+
 	// Remove IP from POD interface
-	err := Handler.AddrDel(b.podNicLink, &b.vif.IP)
+	if delegateIp {
+		err := Handler.AddrDel(b.podNicLink, &b.vif.IP)
 
-	if err != nil {
-		log.Log.Reason(err).Errorf("failed to delete link for interface: %s", podInterface)
-		return err
+		if err != nil {
+			log.Log.Reason(err).Errorf("failed to delete link for interface: %s", podInterface)
+			return err
+		}
 	}
 
 	// Set interface link to down to change its MAC address
-	err = Handler.LinkSetDown(b.podNicLink)
+	err := Handler.LinkSetDown(b.podNicLink)
```

Comment:
I'm just wondering why do you think we should randomize the mac address of the pod even when delegateIp is false?

Reply:
That's partially because we haven't solved custom MAC address for OpenShift SDN yet. If we don't move the allocated MAC address to VIF (and hence replace it on br1 with a random one) then no traffic coming from the guest will pass rigid ARP flows the SDN sets specific to the allocated MAC address.

Maybe we should leave it as-is and wait until the issue is solved for the SDN in general.